robots/pkg/kubevirt/cmd/remove/jobs_test.go

```diff
@@ -0,0 +1,555 @@
+											JobStatesToReport: []v1.ProwJobState{},
+										},
+									},
+									Name: jobconfig.CreatePresubmitJobName(newMinorSemver("1", "19"), "sig-network"),
+									Spec: &corev1.PodSpec{
+										Containers: []corev1.Container{
+											{
+												Env: []corev1.EnvVar{},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				targetRelease: newMinorSemver("1", "19"),
+			},
+			wantUpdated: true,
+			wantJobConfig: &config.JobConfig{
+				PresubmitsStatic: map[string][]config.Presubmit{
+					jobconfig.OrgAndRepoForJobConfig: {
+						{
+							JobBase: config.JobBase{
+								Labels: map[string]string{},
+								ReporterConfig: &v1.ReporterConfig{
+									Slack: &v1.SlackReporterConfig{
+										JobStatesToReport: []v1.ProwJobState{},
+									},
+								},
+								Name: jobconfig.CreatePresubmitJobName(newMinorSemver("1", "20"), "sig-network"),
+								Spec: &corev1.PodSpec{
+									Containers: []corev1.Container{
+										{
+											Env: []corev1.EnvVar{},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := deletePresubmitJobsForRelease(tt.args.jobConfig, tt.args.targetRelease); got != tt.wantUpdated {
+				t.Errorf("deletePresubmitJobsForRelease() = %v, wantUpdated %v", got, tt.wantUpdated)
+			}
+			if tt.wantUpdated && !reflect.DeepEqual(tt.args.jobConfig, tt.wantJobConfig) {
+				t.Errorf("deletePresubmitJobsForRelease() = %v", deep.Equal(tt.args.jobConfig, tt.wantJobConfig))
+			}
+		})
+	}
+}
+
+func Test_removeOldJobsIfNewOnesExist(t *testing.T) {
+	type args struct {
+		releases []*github.RepositoryRelease
+		removeJobsOpts removeJobsOptions
+	}
+	tests := []struct {
+		name string
+		args args
+		wantModification bool
+	}{
+		{
+			name: "should modify",
+			args: args{
+				releases: []*github.RepositoryRelease{
+					newRelease("v1.22.0"),
+					newRelease("v1.21.0"),
+					newRelease("v1.20.0"),
+					newRelease("v1.19.0"),
+				},
+				removeJobsOpts: removeJobsOptions{
+					jobConfigPathKubevirtPeriodics: "testdata/should_modify/kubevirt-periodics.yaml",
+					jobConfigPathKubevirtPresubmits: "testdata/should_modify/kubevirt-presubmits.yaml",
+				},
+			},
+			wantModification: true,
+		},
+		{
+			name: "should not modify",
+			args: args{
+				releases: []*github.RepositoryRelease{
+					newRelease("v1.22.0"),
+					newRelease("v1.21.0"),
+					newRelease("v1.20.0"),
+					newRelease("v1.19.0"),
+				},
+				removeJobsOpts: removeJobsOptions{
+					jobConfigPathKubevirtPeriodics: "testdata/should_not_modify/kubevirt-periodics.yaml",
+					jobConfigPathKubevirtPresubmits: "testdata/should_not_modify/kubevirt-presubmits.yaml",
+				},
+			},
+			wantModification: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tempDir, err := os.MkdirTemp("", "removeOldJobsIfNewOnesExist-")
```

Comment:
Not for this PR, but we might consider migrating to something like afero to abstract fs dependency in unit tests https://github.com/spf13/afero